rfcs/20221124-monorepo.md

```diff
@@ -0,0 +1,179 @@
+↖️ Table of Contents
+
+# RFC: Monorepo
+
+**Status:** 🚧 WIP, comments are welcome nonetheless
+
+## Reviewers
+
+- [ ] @zackkrida
+- [ ] @sarayourfriend
+
+## Rationale
+
+For a comprehensive discussion about the pros, the cons and the counterpoints to each see [discussion](https://github.com/WordPress/openverse/issues/192). This is not the purpose of this RFC.
+
+This RFC summarily lists the benefits and then, with the twin assumptions of a monorepo being ultimately beneficial and the decision to migrate being finalised in the above discussion, proceeds to go into the implementation details.
+
+### Exclusive benefits of monorepo
+
+This only includes things that cannot be accomplished without the use of a monorepo.
+
+1. Single place to go for issues, PRs and all activity. Currently tickets are scattered across several repos, and any tickets that could benefit more than a single layer must be opened in each of the different repos.
+
+1. Singular copy (different from synced independent copies) of scaffolding code such as Git hooks, lint rules and common workflows. This is distinctly better than elaborate sync workflows.
+
+1. Central place for all technical documentation, enabling documentation for different parts of the stack to cross-reference other pieces and stay current with changes in other places.
+
+1. Enables the infra to deploy the code to coexist with the code itself. Apart from the private secrets that will still need to be encrypted, the IaC files could be organised identical to the code.
+
+1. Milestones that can span across multiple layers of the stack are only possible in GitHub. This is a limitation imposed by GitHub and there is no workaround for this.
```

Comment:
I see manual meta issues as good enough workaround.

Reply:
> I see manual meta issues as good enough workaround.

They definitely might be. We would need to establish some conventions and patterns for using them more effectively, but GitHub _did_ recently fix some bugs with showing the titles of issues from other repos. Here's an example of a cross-repo meta issue:

https://github.com/WordPress/openverse/issues/341

It's really nice! Meta issues are also much more flexible than milestones, as milestones do not allow for comments or discussion. Regardless of switching to a monorepo or not we might want to use meta issues more _anyway_.

One _minor_ benefit to the monorepo is that you get autocompletion of issues in the current repo, but not for cross-repo issues: 

![CleanShot 2022-12-06 at 12 12 32](https://user-images.githubusercontent.com/6351754/205977842-9d8ce4f2-1bcf-4c59-ac55-6e2f9990862c.png)